### PR TITLE
Move reference of infra config from annotation to spec schema

### PR DESCRIPTION
**Pull request by lulf**
This allows the existence of the infraconfig field to be validated by OpenAPI schema
validation, as well as allowing the Operator Catalog UI to reference the
field.

### CODE REVIEW

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=plans.PlansTest

File: templates/crds/addressspaceplans.crd.yaml [hunk 1]
```diff
@@ -32,6 +32,7 @@ spec:
             - addressSpaceType
             - resourceLimits 
             - addressPlans
+            - infraConfig
```
Comment (k-wall) [inline templates/crds/addressspaceplans.crd.yaml]: Should this be `infraConfigRef`?

File: templates/crds/addressspaceplans.crd.yaml [hunk 2]
```diff
@@ -43,6 +44,13 @@ spec:
               type: string
             addressSpaceType:
               type: string
+            infraConfigRef:
```
Comment (k-wall) [inline templates/crds/addressspaceplans.crd.yaml]: Here we declare an object, but later, if the example plans for instance, it is used as if it were a string.  How does this work?

Comment (lulf) [reply templates/crds/addressspaceplans.crd.yaml]: Fixed

**Comment by kornys** [comment]
@lulf this will require update in systemtests as well, right?

**Comment by lulf** [comment]
@kornys It will anyway fall back to the annotation, but I have updated the systemtest code after the merge of using api-model

**Pull request merged**